src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisShardSyncer.java

```diff
@@ -545,7 +492,7 @@ private static ShardFilter getShardFilterAtInitialPosition(InitialPositionInStre
      * @return true if the shard is a descendant of any current shard (lease already exists)
      */
     // CHECKSTYLE:OFF CyclomaticComplexity
-    boolean checkIfDescendantAndAddNewLeasesForAncestors(String shardId,
+    static boolean checkIfDescendantAndAddNewLeasesForAncestors(String shardId,
```

Comment:
We can make `LeaseSynchronizer` et al. to be inner classes/interfaces of `KinesisShardSyncer` to avoid exposing this, I don't think any other classes will need to access this.

Reply:
I extracted these into separate classes to keep KinesisShardSyncer file manageable, but I agree that it ended up exposing a lot of internal methods. I'll move these back into inner classes.

Reply:
Yeah sounds fine